From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Wed, 2 Sep 2026 10:34:58 +0200
Subject: fix(dracut): correct order of "Creating image" log line

The log line "Creating image" is followed by "Hardlinking files":

```
[...]
dracut[I]: *** Including modules done ***
dracut[I]: *** Installing kernel module dependencies ***
dracut[I]: *** Installing kernel module dependencies done ***
dracut[I]: *** Resolving dependencies for executables and libraries ***
dracut[I]: *** Resolving dependencies for executables and libraries done ***
dracut[I]: *** Generating early-microcode cpio image ***
dracut[I]: *** Constructing GenuineIntel.bin ***
dracut[I]: *** Store current command line parameters ***
dracut[I]: *** Stripping files ***
dracut[I]: *** Stripping files done ***
dracut[I]: *** Creating image file '/var/tmp/dracut-test.doZhty/initramfs.testing' ***
dracut[I]: *** Hardlinking files ***
dracut[I]: *** Hardlinking files done ***
dracut[I]: Using auto-determined compression method 'zstd'
dracut[I]: *** Creating initramfs image file '/var/tmp/dracut-test.doZhty/initramfs.testing' done ***
```

Move the "Creating image" log line back to where it belonged:

```
[...]
dracut[I]: *** Including modules done ***
dracut[I]: *** Installing kernel module dependencies ***
dracut[I]: *** Installing kernel module dependencies done ***
dracut[I]: *** Resolving dependencies for executables and libraries ***
dracut[I]: *** Resolving dependencies for executables and libraries done ***
dracut[I]: *** Generating early-microcode cpio image ***
dracut[I]: *** Constructing GenuineIntel.bin ***
dracut[I]: *** Store current command line parameters ***
dracut[I]: *** Stripping files ***
dracut[I]: *** Stripping files done ***
dracut[I]: *** Hardlinking files ***
dracut[I]: *** Hardlinking files done ***
dracut[I]: *** Creating image file '/var/tmp/dracut-test.nfRABi/initramfs.testing' ***
dracut[I]: Using auto-determined compression method 'zstd'
dracut[I]: *** Creating initramfs image file '/var/tmp/dracut-test.nfRABi/initramfs.testing' done ***
```

Fixes: 9fdf683f6d2 ("fix(dracut): ensure hardlink deduplication is reproducible")
Origin: upstream, https://github.com/dracut-ng/dracut/commit/1de52aca55cf16bc469b65b12f91e75e9a252b58
---
 dracut.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dracut.sh b/dracut.sh
index 8f2da41..5f149e0 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -3136,8 +3136,6 @@ if [[ -e $outfile ]]; then
     outfile="${outfile}.tmp"
 fi
 
-dinfo "*** Creating image file '$outfile' ***"
-
 # Read list of files and echo them plus all leading directories.
 # The same directories might be printed multiple times (even with sorted input)!
 add_directories() {
@@ -3206,6 +3204,8 @@ if [[ $do_hardlink == yes ]] && command -v hardlink > /dev/null; then
     fi
 fi
 
+dinfo "*** Creating image file '$outfile' ***"
+
 [[ $EUID != 0 ]] && cpio_owner="0:0"
 
 [[ $EUID == 0 ]] || owner_override="\t\t\t0\t0"
